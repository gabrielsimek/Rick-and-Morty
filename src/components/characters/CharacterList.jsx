import React from 'react';
import PropTypes from 'prop-types';
import CharacterItem from './CharacterItem';

const CharacterList = ({ characters }) => {
  const charactersList = characters.map(character => {
    return (
      <li key={character.id}>
        <CharacterItem 
          {...character}         
        />
      </li>
    );
  });


  return <ul>{charactersList}</ul>;

};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      species: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired

    })
  )
};

export default CharacterList;
