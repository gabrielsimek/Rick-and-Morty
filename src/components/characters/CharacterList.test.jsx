import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';
describe('App CharacterList', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<MemoryRouter><CharacterList characters={
      [{ name: 'name', species: 'species', status: 'status', image: 'image' }]
    }/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
