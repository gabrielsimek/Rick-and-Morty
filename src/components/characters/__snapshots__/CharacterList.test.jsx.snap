// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App CharacterList renders App 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/character/undefined"
      >
        <figure>
          <img
            src="image"
          />
          <figcaption>
            <h2>
              name
            </h2>
            <p>
              species
            </p>
            <p>
              status
            </p>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
